src/mrpro/data/traj_calculators/KTrajectoryPulseq.py

```diff
@@ -54,13 +54,17 @@ def __call__(self, kheader: KHeader) -> KTrajectoryRawShape:
             raise ValueError('We currently only support constant number of samples')
         n_k0 = int(n_samples.item())
 
-        def reshape_pulseq_traj(k_traj: torch.Tensor, encoding_size: int):
-            k_traj *= encoding_size / (2 * torch.max(torch.abs(k_traj)))
+        def rescale_and_reshape_traj(k_traj: torch.Tensor, encoding_size: int):
+            if encoding_size > 1 and torch.max(torch.abs(k_traj)) > 0:
+                scaling_factor = encoding_size / (2 * torch.max(torch.abs(k_traj)))
+            else:
+                scaling_factor = 0
+            k_traj *= scaling_factor
```

Comment:
dont do inplace here. 
this only made debugging annoying, as running line 66ff changes k_traj_adc ...